### PR TITLE
Refactor Hash Password Architecture

### LINKED ISSUE #420: Refactor Hash Password Architecture
**Purpose**
What will this feature provide?

Simplify the current architecture we have for storing/hashing passwords. 

**Proposed Solution**
If you have an idea on how to implement this feature, please share!

Currently hashing methods are stored in /utils/auth.py and are imported to /resources/user.py to be used for the register, login, and patch endpoints. It also has to be imported to multiple files that deal with testing user endpoints.

To simplify it - the methods can be migrated directly into the User model, so hashing and checking can occur in one place.

**Acceptance Criteria**
Please list the requirements the implementation of this feature should meet.

N/A

**Mockup Screenshot or Page Number**
Paste a screenshot of the feature from the FIGMA mockups or cite the page number.

N/A

**Additional context**

None

### PR DESCRIPTION
**Pull request by cecillesalazar**
### What issue is this solving?

Closes codeforpdx/dwellingly-app/issues/420

### Any helpful knowledge/context for the reviewer?

Nope, just a refactor - simplifying the current structure of hashing/storing passwords

Is a re-seeding of the database necessary?

Nope

Any new dependencies to install?

Nope

Any special requirements to test?

Nope

### Please make sure you've attempted to meet the following coding standards
- [x] Code builds successfully
- [x] Code has been tested and does not produce errors
- [x] Code is readable and formatted
- [x] There isn't any unnecessary commented-out code

### CODE REVIEW

**Review by NickSchimek — CHANGES_REQUESTED** [review]
This is great. This handles automatic password hashing when the object is instantiated. However, we also need it for when an object is updated. I pushed up a missing test to expose it. One possible solution is to create getter and setters using pythons built-in decorators.

Something like this? Or do you have another idea?
```
def __init__(self, ...
    ...
    self._password = UserModel.hash_pw(password)
    ...


@property
def password(self):
   return self._password

@password.setter
def password(self, value):
   self._password = UserModel.hash_pw(value)
```

**Comment by cecillesalazar** [comment]
@NickSchimek - nice catch. Test should be passing now - branch is rebased too~

File: models/user.py [hunk 1]
```diff
@@ -68,7 +76,7 @@ def hash_pw(plaintext_password):
         return bcrypt.hashpw(bytes(plaintext_password, 'utf-8'), bcrypt.gensalt())
 
     def check_pw(self, plaintext_password):
-        return bcrypt.checkpw(bytes(plaintext_password, 'utf-8'), self.password)
+        return bcrypt.checkpw(bytes(plaintext_password, 'utf-8'), self._password)
```
Comment (NickSchimek) [inline models/user.py:79]: small detail: I think we should keep this as `self.password` which does end up returning `self._password`, however it is best practice to use getter methods for accessing private fields.

**Review by NickSchimek — CHANGES_REQUESTED** [review]
one tiny suggestion.

**Review by NickSchimek — APPROVED** [review]
This is great! Thanks for updating this. I will go ahead and make that minor change in a PR to speed up the test suite. I will request your review.

**Pull request merged**